### ISSUE TITLE
Gems with native extensions aren't packed

### ISSUE THREAD

**Issue by bryanp** [pmq20/ruby-packer#122]
If my `Gemfile` includes a gem with a native extension, `rubyc` generates an output but the output fails to run. I'm on macOS 10.15.6 running Ruby Packer `v2.7.1a`.

Here's an example `Gemfile`:

```ruby
source "https://rubygems.org"

gem "nio4r"
```

And here's the entrance file:

```ruby
#!/usr/bin/env ruby
require "rubygems"
require "bundler/setup"
Bundler.require(:default)
```

I'm using `rubyc` like this:

```
rubyc --tmpdir=../build --keep-tmpdir entrance
```

Here's the error when running `a.out`:

```
$ ./a.out
Could not find nio4r-2.5.2 in any of the sources
Run `bundle install` to install missing gems.
```

The Rails example in the `README` builds correctly but fails to run, throwing missing gem errors for dependencies with native extensions. I also tried using other gems with native extensions, such as `ffi` and `sassc` with the same result as `nio4r`. I have successfully built projects that _don't_ have native gem dependencies.

I do see one error at the end of `rubyc`, but I don't know that it's related to this:

```
*** Following extensions are not compiled:
gdbm:
	Could not be configured. It will not be installed.
```

I've tried to resolve this error but no luck. Here's what's at the top of `gdbm/mkmf.log`:

```
have_library: checking for gdbm_open() in -lgdbm... -------------------- no
```

I confirmed that `gdbm` is installed with homebrew:

```
$ brew list | grep gdbm
gdbm
```

Could this point to some underlying issue on my system that would cause native gem extensions to fail? 

Should I even expect gems with native extensions to be packed correctly?

**Comment by pmq20** [comment]
@bryanp Hi Bryan, thanks for reporting the issue! Yes, we expect gems with native extensions to be packed correctly. I'll reproduce it and I also plan to set up a CI runner to guarantee that native gems work successfully. Due to work, I have to work on it in my next weekend. Stay tuned :)

**Comment by bryanp** [comment]
@pmq20 is there anything I can do to help troubleshoot this? I don't mind pitching in, but am not sure where to start.

**Comment by pmq20** [comment]
@bryanp Thank you Bryan for offering help! This issue might be related to #123. The same characteristic of those issues is that the require process has problems (bugs). As I also said in #123, I think the way to go is to add more debug output to the require process to diagnose the problem.

That said, this issue might be easier to reproduce than #123, because `nio4r` has C extensions and I have not been actively testing `nio4r` recently, nor is it part of CI yet. But it should work because the design is to extract the extension from the packed memfs to a temporary location and `dlopen` it. Due to weekday work, I have to look at it in my weekends.

**Comment by bryanp** [comment]
Thanks! I'll see if I can find time to help contribute here. I appreciate your response, totally understand limited availability 😃

**Comment by skyeagle** [comment]
This issue has to do with #75. I've just verified that by building `rubyc` from the source with removed `so` and `bundle` items from the list here https://github.com/pmq20/ruby-packer/blob/master/lib/compiler.rb#L314 and it worked. Although, I'm experiencing another issue with the packer at this moment.

**Comment by bryanp** [comment]
@skyeagle Hmm that seems likely to be the problem. I tried testing this out by building rubyc from source (latest master) but hitting other issues. It builds just fine, but running it results in an error like this:

```
Could not find ansi-1.5.0 in any of the sources
Run `bundle install` to install missing gems.
```

So I'm not entirely sure what's going on there. @pmq20 any ideas on why my local build would fail in this way?

**Comment by bryanp** [comment]
Ah, actually this seems like #123 and is probably something environment-specific.

**Comment by bryanp** [comment]
This can be closed now that #132 is merged.
